src/controllers/users.js

```diff
@@ -26,6 +30,63 @@ class userController {
       'Last Name': lastName,
     });
   }
-}
 
+  static async getuserProfile(req, res) {
+    const user = await User.findOne({ userName: req.params.username });
+    if (!user) return res.status(400).json({ message: 'User not found' });
+    return res.status(200).json({
+      _id: String(user._id),
+      userName: user.userName,
+      email: user.email,
+      firstName: user.firstName,
+      lastName: user.lastName,
+      socialMedia: user.socialMedia,
+      tel: user.tel,
+      bio: user.bio,
+      website: user.website,
+    });
+  }
+
+  static async updateUserProfile(req, res) {
+    const user = await User.findOne({ userName: req.params.username });
+
+    if (String(user._id) !== req.user.userId)
+      return res.status(401).json({ message: 'Not authorized' });
+
+    const { email, firstName, lastName, socialMedia, tel, bio, website } =
+      req.body;
+
+    const updatedProfile = await User.findByIdAndUpdate(
+      { _id: String(user._id) },
+      {
+        $set: {
+          email,
+          firstName,
+          lastName,
+          socialMedia,
+          tel,
+          bio,
+          website,
+        },
+      },
+      { new: true }
+    );
+
+    return res.status(200).json({
+      _id: updatedProfile._id,
+      userName: updatedProfile.userName,
+      email: updatedProfile.email,
+      firstName: updatedProfile.firstName,
+      lastName: updatedProfile.lastName,
+      socialMedia: updatedProfile.socialMedia,
+      tel: updatedProfile.tel,
+      bio: updatedProfile.bio,
```

Comment:
Destructuring this would actually create more lines of code. 
Destructuring the user object means we lose access to the object properties that would make destructuring advantageous. This means that id we destruture the updateProfile object, we are going to write more code.

Basically withouth destructuring we would have:


return res.status(200).json({
      _id: updatedProfile._id,
      userName: updatedProfile.userName,
      email: updatedProfile.email,
      firstName: updatedProfile.firstName,
      lastName: updatedProfile.lastName,
      socialMedia: updatedProfile.socialMedia,
      tel: updatedProfile.tel,
      bio: updatedProfile.bio,
      website: updatedProfile.website,
    });

But if we destructure we would have:
const {
      _id: updatedId,
      username: updatedUserName,
      email: updatedEmail,
      firstName: updatedFirstName,
      lastName: updatedLastName,
      socialMedia: updatedSocialMedia,
      tel: updatedTel,
      bio: updatedBio,
      website: updatedWebsite,
    } = updatedProfile;
    return res.status(200).json({
      _id: updatedId,
      userName: updatedUserName,
      email: updatedEmail,
      firstName: updatedFirstName,
      lastName: updatedLastName,
      socialMedia: updatedSocialMedia,
      tel: updatedTel,
      bio: updatedBio,
      website: updatedWebsite,
    });